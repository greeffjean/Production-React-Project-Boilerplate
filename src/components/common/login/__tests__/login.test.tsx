
import React from 'react';
import renderer from 'react-test-renderer';
import { render, screen } from "@testing-library/react";
import LoginComponent from '../login';


test('Renders correctly', () => {
    const tree = renderer.create(<LoginComponent greeting={'Jean'} />).toJSON()
    expect(tree).toMatchSnapshot();
})

test("Component renders with correct UI", () => {
    render(<LoginComponent greeting={'Jean'} />);
    const userGreeting = screen.getByRole('userGreeting');
    expect(userGreeting).toHaveTextContent('Jean');
    expect(userGreeting).toHaveAttribute('role', 'userGreeting');
  });

