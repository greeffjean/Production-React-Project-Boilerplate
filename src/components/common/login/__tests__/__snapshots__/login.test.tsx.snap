// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
Array [
  <h1
    role="welcomeHeading"
  >
    This is the Login Page!
  </h1>,
  <span
    role="userGreeting"
  >
    Jean
  </span>,
]
`;
